### ISSUE TITLE
Uncaught TypeError: Data must be a string or a buffer

### ISSUE THREAD

**Issue by littleskunk** [storj-archived/kfs#36]
One storjshare GUI user (OSX 10.12.1) is not sending any OFFER message and developer console has this error messages. Every time he would like to send an OFFER he will write a new error message.

Different GUI versions and start farming with an empty folder didn't help. Still this error message:
```
Uncaught TypeError: Data must be a string or a bufferHash.update @ crypto.js:73
module.exports.hashKey @ utils.js:45
module.exports.coerceKeyOrIndex @ utils.js:60
_getStat @ b-table.js:193
(anonymous function) @ async.js:1156
replenish @ async.js:1030
(anonymous function) @ async.js:1034
_asyncMap @ async.js:1154
(anonymous function) @ async.js:1240
Btable.stat @ b-table.js:213
(anonymous function) @ embedded.js:173
```

**Comment by ghost** [comment]
The problem is only on external HDD. If I use internal SSD no issues.

External HDD is connected directly via USB, in HFS+

**Comment by ghost** [comment]
Overnight, same settings, my internal HD received 5,5GB. My external, 0b

<img width="567" alt="screenshot 2016-11-05 11 06 12" src="https://cloud.githubusercontent.com/assets/22933729/20029594/ed40c02a-a347-11e6-88eb-068d46b9b57d.png">

**Comment by ghost** [comment]
Correction: a new drive on the same External HDD is working, receiving contracts, and increasing used space

**Comment by littleskunk** [comment]
Reproduceable with CLI. Start farming with a new storage location and nodejs 6.9.1 and you will get this error. Do the same with nodejs 4.6.1 and everything is running fine.

Workaround:
CLI: Install nodejs 4.6.1
GUI: Install GUI v2.0.16

@bookchin kfs is not working with nodejs 6.9.1 even if travis has no unit test for this special case.

**Comment by littleskunk** [comment]
nodejs 4.6.1 survived a few minutes longer but at the end the same error.

```
crypto.js:70
 this._handle.update(data, encoding);
              ^

TypeError: Not a string or buffer
   at TypeError (native)
   at Hash.update (crypto.js:70:16)
   at Object.module.exports.hashKey (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/lib/utils.js:45:44)
   at Object.module.exports.coerceKeyOrIndex (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/lib/utils.js:60:28)
   at _getStat (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/lib/b-table.js:193:21)
   at /Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/node_modules/async/dist/async.js:1156:9
   at replenish (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/node_modules/async/dist/async.js:1030:17)
   at /Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/node_modules/async/dist/async.js:1034:9
   at _asyncMap (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/node_modules/async/dist/async.js:1154:5)
   at Object.mapLimit (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/node_modules/async/dist/async.js:1240:16)
   at Btable.stat (/Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/kfs/lib/b-table.js:213:9)
   at /Users/user/.nvm/versions/node/v4.6.1/lib/node_modules/storjshare-cli/node_modules/storj-lib/lib/storage/adapters/embedded.js:173:16
```

**Comment by ghost** [comment]
I'm using GUI v2.0.15 now, and the new drive on external HDD is working, the internal SSD is working, and the "old" aka problematic drive is not working still, like before... Install GUI v2.0.15 didn't fix the problem.

**Comment by littleskunk** [comment]
Fixed in KFS v2.3.0. As soon as I am at home I will build new binaries for you :)
